docs/README.md

```diff
@@ -38,181 +38,92 @@ Code | Text | Description
 
 ### Headers and request signing
 
-All API calls need to be signed using HMAC and SHA256 or SHA512.
+All API calls need to be signed using HMAC and SHA-256 or SHA-512. When a request contains a body, the body must be valid JSON and `content-type` header with value `application/json; charset=utf-8` should be included. Other encodings may work too, but using UTF-8 is preferred.
 
-HMAC is calculated from the HTTP request's body payload using the signing secret. Header keys must be sorted alphabetically, lower-cased and included in HMAC calculation.
+All API responses are signed the same way allowing merchant to verify response validity. In addition the reponses contain `cof-request-id` header. Saving or logging the value of this header is recommended.
 
-* The merchant ID needs to be in `Checkout-Account`
-* The algorithm needs to be in `Checkout-Algorithm` (SHA256 or SHA512)
-* The calculated HMAC needs to be sent in each request in the HTTP header `Checkout-Signature`
-* The method header `Checkout-Method` needs to be set to `POST`, `GET` or `DELETE`
+The signature is transmitted in `signature` HTTP header. Signature payload consists the following fields separated with a line feed (\n). Carrige returns (\r) should not be used.
 
-So the signature is calculated from these values as in this example:
-
-```
-checkout-account:1234\n
-checkout-algorithm:SHA512\n
-checkout-method:POST\n
-checkout-nonce:1234\n
-checkout-timestamp:2018-07-05T11:19:25.950Z\n
-REQUEST BODY
-```
-
-These are passed to the HMAC function which uses SHA256 algorithm. Carriage returns (\r) should not be used, only line feed (\n).
+* All `checkout-` headers in alphabetical order. The header keys must be in lowercase. Each header key and value are separated with `:`
+* HTTP body, or empty string if no body
 
+The headers are:
 
 field | info | description
 --- | --- | ---
-checkout-account | numeric | Checkout account ID
-checkout-algorithm | string | Used algorithm, either SHA256 or SHA512
-checkout-method | string | HTTP verb of request
+checkout-account | numeric | Checkout account ID, eg. 375917
+checkout-algorithm | string | Used signature algorithm, either sha256 or sha512
+checkout-method | string | HTTP verb of request, either GET or POST
 checkout-nonce | string | Unique identifier for this request
 checkout-timestamp | string | ISO 8601 date time
+checkout-transaction-id | string | Checkout transaction ID when accessing single transaction - not required for new payment request
 
-The HTTP verb, nonce and timestamp are used to mitigate various replay and timing attacks.
-
-### HMAC example (node.js)
+The HTTP verb, nonce and timestamp are used to mitigate various replay and timing attacks. Below is an example of the payload passed to a HMAC function:
```

Comment:
not sure if the meaning stays the same this way, but "below is an example payload" would maybe sound more fluent or something